editor/src/scene_viewer/gizmo.rs

```diff
@@ -49,6 +50,10 @@ pub struct SceneGizmo {
     pub pos_z: Handle<Node>,
     pub neg_z: Handle<Node>,
     pub center: Handle<Node>,
+    pub first_yaw: f32,
+    pub first_pitch: f32,
+    pub is_dragging: bool,
+    pub first_click_pos: Vector2<f32>,
```

Comment:
I was thinking of grouping them in a struct called SceneGizmoOrbital. Something like
``` rust 
struct SceneGizmoOrbital {
    pub initial_click_pos: Vector2<f32>,
    pub is_dragging: bool,
    pub gizmo_rotation: CameraRotation, 
}
```

Reply:
You can do this:

```rust
struct DragContext {
    pub initial_click_pos: Vector2<f32>,
    pub initial_rotation: CameraRotation, 
}
```

and in the gizmo:

```rust
drag_context: Option<DragContext>
```

So you don't need to have `is_dragging` flag, because you can do `if let Some(drag_context) = self.drag_context.as_ref() {..}`.